libgnucash/engine/gnc-commodity.cpp

```diff
@@ -1830,11 +1819,9 @@ gnc_commodity_table_insert(gnc_commodity_table * table,
     nsp = gnc_commodity_table_add_namespace(table, ns_name, book);
 
     PINFO ("insert %p %s into nsp=%p %s", priv->mnemonic, priv->mnemonic,
-           nsp->cm_table, nsp->name);
-    g_hash_table_insert(nsp->cm_table,
-                        (gpointer)CACHE_INSERT(priv->mnemonic),
-                        (gpointer)comm);
-    nsp->cm_list = g_list_append(nsp->cm_list, comm);
+           &nsp->cm_table, nsp->name);
+    nsp->cm_table[priv->mnemonic] = comm;
```

Comment:
this removal of CACHE_INSERT should cause refcount error but in practice it doesn't...